es_ES/hot-reload.md

```diff
@@ -0,0 +1,311 @@
+
+Si los cambios de código afectan al estado de tu app (o sus dependencias),
+los datos que tiene tu app para trabajar podrían no ser totalmente 
+consistentes con los datos que deberían tener si se ejecutan desde 
+el principio. El resultado podría ser un comportamiento diferente
+después de un hot reload que con un hot restart.
+
+Por ejemplo, si modificas la definición de una clase para heredar desde StatelessWidget
+a StatefulWidget (o viceversa), después de hacer hot reload el estado previo de tu 
+app es conservado. Sin embargo, el estado podría no ser compatible con 
+los nuevos cambios.
+
+Considera el siguiente código:
+
+```
+class myWidget extends StatelessWidget {
+  Widget build(BuildContext context) {
+    return GestureDetector(onTap: () => print('T'));
+  }
+}
+```
+Después de ejecutar la app, si haces el siguiente cambio:
+
+```
+class myWidget extends StatefulWidget {
+  @override
+  State createState() => myWidgetState();
+}
+class myWidgetState {
+...
+...
+}
+```
+
+y entonces haces hot reload, la consola muestra un _assertion failure_ parecido a:
+
+```
+myWidget is not a subtype of StatelessWidget
+```
+
+Es estas situaciones, es necesario un hot restart para ver la app actualizada.
+
+## Cambio del código reciente se incluye pero el estado de la app se excluye
+
+En Dart, [los campos estáticos son inicializados de forma perezosa](https://news.dartlang.org/2012/02/static-variables-no-longer-have-to-be.html). Esto significa que la 
+primera vez que ejecutas una app Flutter y un campo estático es leído, este se ajusta al valor que sea 
+que marque la evaluación de su inicializador.
+Las variables Globales y los campos estáticos son tratados como estados, y por lo tanto 
+no se reinicializan durante un hot reload.
+
+Si cambias los inicializadores de una variable global o de un campo estático, es necesario
+reiniciar por completo para ver estos cambios. Por ejemplo, considera el 
+siguiente código:
+
+```
+final sampleTable = [
+  Table("T1"),
+  Table("T2"),
+  Table("T3"),
+  Table("T4"),
+];
+```
+Despues de ejecutar la app, si haces el siguiente cambio:
+```
+final sampleTable = [
+  Table("T1"),
+  Table("T2"),
+  Table("T3"),
+  Table("T10"),    //modificado
+];
+```
+y entonces haces hot reload, el cambio no se refleja.
+
+A la inversa, en el siguiente ejemplo:
+```
+const foo = 1;
+final bar = foo;
+void onClick(){
+  print(foo);
+  print(bar);
+}
+```
+ejcutar la app por primera vez imprime `1` y `1`. Enconces si haces el siguiente
+cambio:
+
+```
+const foo = 2;    //modificado
+final bar = foo;
+void onClick(){
+  print(foo);
+  print(bar);
+}
+```
+y haces hot reload, ahora imprime `2` y `1`. Mientras que los de valor para campos `const` 
+son siempre recargados, el inicializador del campo estático no se re-ejecuta.
+Conceptualmente, los campos `const` son tratados como alias en lugar de estados.
+
+La VM Dart detecta los cambios en inicializadores y etiqueta cuando un conjunto 
```

Comment:
etiquetas (plural)

Reply:
The Dart VM detects initializer changes and flags when a set of changes needs a hot restart to take effect. The flagging mechanism is triggered for most of the initialization work in the above example, but not for cases like:

Yo interpreto que la máquina virtual Dart detectas los cambios de los inicializadores y entonces "etiqueta" cuando un conjunto de cambios .....